craft_parts/errors.py

```diff
@@ -168,10 +171,13 @@ def from_validation_error(
         formatted_errors: list[str] = []
 
         for error in error_list:
-            loc = error.get("loc")
-            msg = error.get("msg")
+            try:
+                loc = error["loc"]
+                msg = error["msg"]
+            except KeyError:
+                continue
```

Comment:
We could use `get` to default to `None` and try-catch would be unnecessary here, slightly improving readability. The conditional below would handle the absence of keys.